src/guide/typescript-support.md

```diff
@@ -372,6 +372,76 @@ year.value = 2020 // ok!
 If the type of the generic is unknown, it's recommended to cast `ref` to `Ref<T>`.
 :::
 
+### Typing Template Refs
+
+Sometimes you might need to annotate a template ref for a child component in order to call its public method. For example, we have a `MyModal` child component with a method that opens the modal:
+
+```ts
+import { defineComponent, ref } from 'vue'
+
+export default defineComponent({
+  setup() {
+    const isContentShown = ref(false)
+    const open = () => (isContentShown.value = true)
+
+    return {
+      isContentShown,
+      open
+    }
+  }
+})
+```
+
+We want to call this method via a template ref from the parent component:
+
+```html
+<template>
+  <div id="app">
```

Comment:
I tried to run this example myself. It does work but I found myself unsure which style of component is was trying to use. It doesn't seem self-consistent to me.

The `<template>` tag suggests that this is supposed to be read as an SFC. If that's the case, then we should probably go all-in and combine the two code fragments into a single SFC example. In that scenario, the `id=app"` should probably be removed.

An alternative is that this is supposed to be a root-level `App` component, with an in-DOM template. In that case we probably shouldn't have the `<template>` tags around the template. For an in-DOM template we should also use a proper closing tag for `</my-modal>` rather than relying on a self-closing tag.

Reply:
@skirtles-code thank you for teh review! I've changed the example to inline template, could you please take a look?